apps/core/Results.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+
+import * as React from 'react';
+import { View, FlatList } from 'react-native';
+import { QueryRenderer, graphql } from '@kiwicom/margarita-relay';
+import { StyleSheet, Text } from '@kiwicom/universal-components';
+
+import EmptyResults from './EmptyResults';
+import type {
+  ItinerariesSearchInput,
+  ResultsQueryResponse,
+} from './__generated__/ResultsQuery.graphql';
+
+type Props = {|
+  +travelFrom: string,
+  +travelTo: string,
+  +dateFrom: string,
+  +dateTo: string,
+|};
+
+type ItemProps = {|
+  node: ?{|
+    +id: string,
+    +price: ?number,
+    +flyFrom: ?string,
+    +flyTo: ?string,
+    +localDeparture: ?string,
+    +localArrival: ?string,
+  |},
+|};
+
+type ResultItemProps = {|
+  item: ItemProps,
+|};
+
+export default class Results extends React.Component<Props> {
+  resultItem = ({ item }: ResultItemProps) => {
+    if (!item.node) {
+      return null;
+    }
+    const { flyFrom, flyTo, localDeparture, localArrival, price } = item.node;
+    return (
+      <Text
+        style={styles.listItem}
+      >{`${flyFrom} / ${localDeparture} -> ${flyTo} / ${localArrival}  | Price: ${price}`}</Text>
+    );
+  };
+
+  keyExtractor = (item: ItemProps, index: number) => item.node?.id;
+
+  renderInner = (props: ResultsQueryResponse) => {
+    const { searchItineraries } = props;
+    if (
+      !searchItineraries ||
+      !searchItineraries.edges ||
+      searchItineraries.edges.length === 0
+    ) {
+      return <EmptyResults />;
+    }
+    return (
+      <>
+        <Text>{`Total results: ${searchItineraries.edges.length}`}</Text>
+        <FlatList
```

Comment:
I think we should move this to a separate component, so renderInner just returns something like `<SearchResultList data={props.searchItineraries} />`

Reply:
Definitely. But I think breaking results to separate components and make results lits better in general is for separate issue.

Reply:
I strongly disagree. This component has way too many responsibilities, and one obvious flaw is the copying of the type `ItemProps`, what when this query changes, we have to maintain this type all the time. 

The main problem here is that this PR is way too big, and you should in the future try to have smaller PR.

Reply:
You're right. The borderline for this issue was little bit unclear and it will be better to split big pieces like this into smaller ones next time. 
Results are now separated into smaller components.